pkg/cli/build.go

```diff
@@ -0,0 +1,416 @@
+		Labels:                  iopts.Label,
+		Layers:                  layers,
+		LogRusage:               iopts.LogRusage,
+		Manifest:                iopts.Manifest,
+		MaxPullPushRetries:      MaxPullPushRetries,
+		NamespaceOptions:        namespaceOptions,
+		NoCache:                 iopts.NoCache,
+		OS:                      systemContext.OSChoice,
+		Out:                     stdout,
+		Output:                  output,
+		BuildOutput:             iopts.BuildOutput,
+		OutputFormat:            format,
+		PullPolicy:              pullPolicy,
+		PullPushRetryDelay:      PullPushRetryDelay,
+		Quiet:                   iopts.Quiet,
+		RemoveIntermediateCtrs:  iopts.Rm,
+		ReportWriter:            reporter,
+		Runtime:                 iopts.Runtime,
+		RuntimeArgs:             runtimeFlags,
+		RusageLogFile:           iopts.RusageLogFile,
+		SignBy:                  iopts.SignBy,
+		SignaturePolicyPath:     iopts.SignaturePolicy,
+		Squash:                  iopts.Squash,
+		SystemContext:           systemContext,
+		Target:                  iopts.Target,
+		TransientMounts:         iopts.Volumes,
+		OciDecryptConfig:        decryptConfig,
+		Jobs:                    &iopts.Jobs,
+		Excludes:                excludes,
+		Timestamp:               timestamp,
+		Platforms:               platforms,
+		UnsetEnvs:               iopts.UnsetEnvs,
+		Envs:                    iopts.Envs,
+		OSFeatures:              iopts.OSFeatures,
+		OSVersion:               iopts.OSVersion,
+	}
+	if iopts.Quiet {
+		options.ReportWriter = ioutil.Discard
+	}
+	return options, containerfiles, nil
+}
+
+func getContainerfiles(files []string) []string {
+	var containerfiles []string
+	for _, f := range files {
+		if f == "-" {
+			containerfiles = append(containerfiles, "/dev/stdin")
+		} else {
+			containerfiles = append(containerfiles, f)
+		}
+	}
+	return containerfiles
+}
+
+// GetFormat translates format string into either docker or OCI format constant
+func GetFormat(format string) (string, error) {
+	switch format {
+	case define.OCI:
+		return define.OCIv1ImageManifest, nil
+	case define.DOCKER:
+		return define.Dockerv2ImageManifest, nil
+	default:
+		return "", errors.Errorf("unrecognized image type %q", format)
+	}
+}
+
+// DecryptConfig translates decryptionKeys into a DescriptionConfig structure
+func DecryptConfig(decryptionKeys []string) (*encconfig.DecryptConfig, error) {
+	decryptConfig := &encconfig.DecryptConfig{}
+	if len(decryptionKeys) > 0 {
+		// decryption
+		dcc, err := enchelpers.CreateCryptoConfig([]string{}, decryptionKeys)
+		if err != nil {
+			return nil, errors.Wrapf(err, "invalid decryption keys")
+		}
+		cc := encconfig.CombineCryptoConfigs([]encconfig.CryptoConfig{dcc})
+		decryptConfig = cc.DecryptConfig
+	}
+
+	return decryptConfig, nil
+}
+
+// EncryptConfig translates encryptionKeys into a EncriptionsConfig structure
+func EncryptConfig(encryptionKeys []string, encryptLayers []int) (*encconfig.EncryptConfig, *[]int, error) {
```

Comment:
You dropped the "get" for this and the prior function and made the availability wider, was that intentional?

Reply:
Yes `Get` is not considered good form in Golang, as I understand it. The functions need to be external for other buildah commands in cmd/buildah/*.go

Reply:
Some or all of these new public functions and constants could conceivably be moved into an `internal` folder so that we don't export them outside of buildah's source tree.